reporter/reporter.go

```diff
@@ -87,25 +88,29 @@ func NewReporter(
 		matcher:         matcher,
 		server:          server,
 		nowFn:           clockOpts.NowFn(),
+		maxPositiveSkew: clockOpts.MaxPositiveSkew(),
 		maxNegativeSkew: clockOpts.MaxNegativeSkew(),
 		metrics:         newReporterMetrics(instrumentOpts.MetricsScope(), instrumentOpts.MetricsSamplingRate()),
 	}
 }
 
 func (r *reporter) ReportCounter(id id.ID, value int64) error {
-	callStart := r.nowFn()
-	multiErr := xerrors.NewMultiError()
-	matchRes := r.matcher.Match(id)
-	earliestNanos := r.nowFn().Add(-r.maxNegativeSkew).UnixNano()
+	var (
+		reportAt  = r.nowFn()
+		fromNanos = reportAt.Add(-r.maxNegativeSkew).UnixNano()
+		toNanos   = reportAt.Add(r.maxPositiveSkew).UnixNano()
+		multiErr  = xerrors.NewMultiError()
+	)
+	matchRes := r.matcher.Match(id, fromNanos, toNanos)
```

Comment:
Not particularly, just feels like it's an important operation that should probably be separated from the rest of the trivial assignments in the var block. Can move if you feel strongly.